cache.go

```diff
@@ -217,21 +217,25 @@ func (a *Cache[K, V]) Clear() {
 	a.policy.Clear()
 }
 
-// Noop if smaller. available should not consider taken space in cache.
+// Noop if smaller. available (+/-) should not consider taken space in cache.
 func (a *Cache[K, V]) SetLargerCapacity(available, max int64) {
+	a.setCapacity(available, max, true)
+}
+
+// available (+/-) should not consider taken space in cache.
+func (a *Cache[K, V]) SetCapacity(available, max int64) {
+	a.setCapacity(available, max, false)
+}
+
+func (a *Cache[K, V]) setCapacity(available, max int64, largerOnly bool) {
 	for {
 		cap := a.cap.Load()
-		size := a.size.Load()
-
-		// If size is over capacity, use capacity as base
-		// to prevent repeated increases even with zero delta.
-		base := min(size, cap)
```

Comment:
How much do we think this is using size instead of cap? Maybe for rollout we could add the SetCapacity path without changing SetLargerCapacity and gradually change to it?

Reply:
Well the place we use SetLargerCapacity is what I would change to SetCapacity anyway. Also it was the size part that caused it to grow even with avail 0.